src/libponyc/pkg/buildflagset.c

```diff
@@ -534,6 +534,35 @@ bool define_build_flag(const char* name)
 }
 
 
+bool remove_build_flags(const char* flags[])
+{
+  pony_assert(flags != NULL);
+
+  if(_user_flags == NULL)
+  {
+    // Initialise flags table.
+    _user_flags = POOL_ALLOC(flagtab_t);
+    flagtab_init(_user_flags, 8);
+  }
+
+  size_t removed = 0;
+  for(const char** next = flags; *next != NULL; next += 1)
+  {
+    flag_t f1 = {stringtab(*next), false};
+    size_t index = HASHMAP_UNKNOWN;
+    flag_t* f2 = flagtab_get(_user_flags, &f1, &index);
```

Comment:
You can replace the `flagtab_get` and `flagtab_removeindex` combination with `flagtab_remove` instead (see here for header definition of `ponyint_hashmap_remove` function: https://github.com/ponylang/ponyc/blob/master/src/libponyrt/ds/hash.h#L83-L88). `flagtab_remove` will return `NULL` if the element wasn't found or else will return the element.  Please also remember to free the element after removing it to avoid a memory leak.

Reply:
ok